### PR TITLE
fix: Ensuring interpolation is disabled for ConfigParser

### LINKED ISSUE #8183: to_ini filter fail to load value with `%` due to default interpolation option.
### Summary

When trying to write an ini file from a dictionary including `%` char in some field, the Filter fail with the following error:
```
invalid interpolation syntax in '<Value>' at position X
```
Looking at the work [here](https://github.com/ansible-collections/community.general/pull/7744), the [default interpolation option](https://docs.python.org/3/library/configparser.html#interpolation-of-values) of `configparser` is still enabled and you would need to escape each `%`.

I feel that this should not be the case, because you are not always in control of the input directory, but also because importing a file using ` | from_ini() ` and writing it back again using ` | to_ini()` is inconsistent.

Additionally, escaping the % (by doubling them) don't produce a single % but a double, making escape impossible.

A simple fix is to disable the interpolation when creating the parent class [here](https://github.com/ansible-collections/community.general/blob/e4e091accaa1d8ecbd7370db6c433bae7a81ce6d/plugins/filter/to_ini.py#L61C1-L67C31) like so:
``` Python3
class IniParser(ConfigParser):
    ''' Implements a configparser which sets the correct optionxform and disable interpolation'''

    def __init__(self):
        super().__init__(interpolation=None)
        self.optionxform = str
```
A more complex fix would require to add an option to enable or not the interpolation.

### Issue Type

Bug Report

### Component Name

filter to_ini

### Ansible Version

```console (paste below)
$ ansible --version
ansible [core 2.15.9]
  config file = /home/vaarlion/.ansible.cfg
  configured module search path = ['/home/vaarlion/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/vaarlion/.local/lib/python3.10/site-packages/ansible
  ansible collection location = /home/vaarlion/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/vaarlion/.local/bin/ansible
  python version = 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] (/usr/bin/python3)
  jinja version = 3.1.2
  libyaml = True
```

### Community.general Version

```console (paste below)
$ ansible-galaxy collection list community.general
# /home/vaarlion/.ansible/collections/ansible_collections
Collection        Version
----------------- -------
community.general 8.3.0 
```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed
CONFIG_FILE() = /home/vaarlion/.ansible.cfg
DISPLAY_SKIPPED_HOSTS(/home/vaarlion/.ansible.cfg) = False
EDITOR(env: EDITOR) = /usr/bin/vim
PAGER(env: PAGER) = less
```

### OS / Environment

Ubuntu 22.04.4 LTS

### Steps to Reproduce

```yaml (paste below)
---
- name: Test % with ini
  hosts: all

  tasks:
    - name: Create a dict with a double % in a value
      ansible.builtin.set_fact:
        test_dict_1:
          test_section:
            a: value a
            b: value %% b

    - name: Display the ini build from the dict using to_ini()
      ansible.builtin.debug:
        msg: "{{ test_dict_1 | community.general.to_ini }}"

    - name: Create a dict with a single % in a value
      ansible.builtin.set_fact:
        test_dict_2:
          test_section:
            a: value a
            b: value % b

    - name: Display the ini build from the dict using to_ini()
      ansible.builtin.debug:
        msg: "{{ test_dict_2 | community.general.to_ini }}"

```

### Expected Results

```console
PLAY [Test % with ini] *******************************************************************

TASK [Gathering Facts] *******************************************************************
ok: [localhost]

TASK [Create a dict with a double % in a value] ******************************************
ok: [localhost]

TASK [Display the ini build from the dict using to_ini()] ********************************
ok: [localhost] => {
    "msg": "[test_section]\na = value a\nb = value %% b\n"
}

TASK [Create a dict with a single % in a value] ******************************************
ok: [localhost]

TASK [Display the ini build from the dict using to_ini()] ********************************
ok: [localhost] => {
    "msg": "[test_section]\na = value a\nb = value % b\n"
}

PLAY RECAP *******************************************************************************
localhost                       : ok=5    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

### Actual Results

```console (paste below)
PLAY [Test % with ini] *******************************************************************

TASK [Gathering Facts] *******************************************************************
ok: [localhost]

TASK [Create a dict with a double % in a value] ******************************************
ok: [localhost]

TASK [Display the ini build from the dict using to_ini()] ********************************
ok: [localhost] => {
    "msg": "[test_section]\na = value a\nb = value %% b\n"
}

TASK [Create a dict with a single % in a value] ******************************************
ok: [localhost]

TASK [Display the ini build from the dict using to_ini()] ********************************
fatal: [localhost]: FAILED! => {"msg": "to_ini failed to parse given dict:invalid interpolation syntax in 'value % b' at position 6. invalid interpolation syntax in 'value % b' at position 6"}

PLAY RECAP *******************************************************************************
localhost                       : ok=4    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   

```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by sscheib**
##### SUMMARY

This PR disables interpolation of ConfigParser and adds test coverage for that. Fixes #8183.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
from_ini
to_ini

### CODE REVIEW

File: changelogs/fragments/8183-from_ini_to_ini.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+bugfixes:
+  - "to_ini - Disabling interpolation of ConfigParser to allow converting values with a %-sign (https://github.com/ansible-collections/community.general/issues/8183)"
+  - "from_ini - Disabling interpolation of ConfigParser to allow converting values with a %-sign (https://github.com/ansible-collections/community.general/issues/8183)"
```
Comment (felixfontein) [inline changelogs/fragments/8183-from_ini_to_ini.yml:3]:
```suggestion
  - "to_ini filter plugin - disabling interpolation of ``ConfigParser`` to allow converting values with a ``%`` sign (https://github.com/ansible-collections/community.general/issues/8183, https://github.com/ansible-collections/community.general/pull/8185)."
  - "from_ini filter plugin - disabling interpolation of ``ConfigParser`` to allow converting values with a ``%`` sign (https://github.com/ansible-collections/community.general/issues/8183, https://github.com/ansible-collections/community.general/pull/8185)."
```

**Review by felixfontein — COMMENTED** [review]
Thanks for fixing this!

**Comment by Vaarlion** [comment]
I didn't find anything wrong with the `from_ini` but i think you are right to disable it everywhere :)

Thank you for that quick fix

**Comment by felixfontein** [comment]
If nobody objects, I'll merge this in a few days.

**Comment by felixfontein** [comment]
@sscheib thanks for your contribution!

**Pull request merged**